slatedb/src/oracle.rs

```diff
@@ -16,19 +20,29 @@ pub(crate) struct Oracle {
     /// The sequence number of the most recent write that has been fully durable
     /// flushed to the remote storage.
     pub(crate) last_remote_persisted_seq: Arc<MonotonicSeq>,
+    /// Tracks the mapping from sequence number to timestamp.
+    pub(crate) seq_tracker: RwLock<TieredSequenceTracker>,
+    /// The system clock (used to track the timestamp of the sequence numbers).
+    system_clock: Arc<dyn SystemClock>,
 }
 
 impl Oracle {
     /// Create a new Oracle with the last committed sequence number. for the read only
     /// db instance (DbReader), only the last committed sequence number is needed to be
     /// tracked, and last_seq and last_remote_persisted_seq are considered to be
     /// the same as last_committed_seq.
-    pub(crate) fn new(last_committed_seq: MonotonicSeq) -> Self {
+    pub(crate) fn new(
+        seq_tracker: TieredSequenceTracker,
+        last_committed_seq: MonotonicSeq,
+        system_clock: Arc<dyn SystemClock>,
```

Comment:
Nit: maybe `with_system_clock` to keep the builder pattern style that this impl uses (and that builder.rs uses)?

Reply:
I considered that, but both seq_tracker and system_clock are "required" to pass in. I don't think there's really good default for it. I can maybe encapsulate the system_clock in the seq_tracker to reduce one argument but I think the builder pattern makes it too easy to miss

Reply:
Fair enough. For internal objects, requiring is OK by me.